### PR TITLE
DOC: create label and ref, fixes broken link

### PR DESCRIPTION
**Pull request by mattip**
continuation of PR #11082, this time with a true label and ref

### CODE REVIEW

File: numpy/lib/npyio.py [hunk 1]
```diff
@@ -475,7 +475,7 @@ def save(file, arr, allow_pickle=True, fix_imports=True):
 
     Notes
     -----
-    For a description of the ``.npy`` format, see :py:mod:`numpy.lib.format`
+    For a description of the ``.npy`` format, see :ref:`binary-serialization`
```
Comment (charris) [inline numpy/lib/npyio.py]: What about the references in `savez` and `savez_compressed`?

Comment (charris) [reply numpy/lib/npyio.py]: Hmm, also need periods at the ends of the sentences in all the places.

Comment (mattip) [reply numpy/lib/npyio.py]: fixed, thanks

**Review by eric-wieser — APPROVED** [review]
LGTM. Any particular reason for now linking to the section not the module? Does it render better?

**Comment by mattip** [comment]
The link didn't work, as you can see on the [devdocs page](http://www.numpy.org/devdocs/reference/routines.io.html), where the **numpy.lib.format** link did not work, even though locally it rendered OK

**Comment by mattip** [comment]
yeah, this doesn't render either. I don't understand why my local builds and the documentation builds are different.

**Comment by charris** [comment]
Are they both using the same numpydoc/sphinx versions?

**Comment by mattip** [comment]
I am checking the circleci log against my local log trying to find the differences.

**Pull request merged**